tests/pauli/test_pauli_arithmetic.py

```diff
@@ -0,0 +1,321 @@
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit Tests for the PauliWord and PauliSentence classes"""
+import pytest
+from copy import copy
+
+from scipy import sparse
+from pennylane import numpy as np
+from pennylane.pauli.pauli_arithmetic import PauliWord, PauliSentence, I, X, Y, Z
+
+
+matI = np.eye(2)
+matX = np.array([[0, 1], [1, 0]])
+matY = np.array([[0, -1j], [1j, 0]])
+matZ = np.array([[1, 0], [0, -1]])
+
+sparse_matI = sparse.eye(2, format="csr")
+sparse_matX = sparse.csr_matrix([[0, 1], [1, 0]])
+sparse_matY = sparse.csr_matrix([[0, -1j], [1j, 0]])
+sparse_matZ = sparse.csr_matrix([[1, 0], [0, -1]])
+
+pw1 = PauliWord({0: I, 1: X, 2: Y})
+pw2 = PauliWord({"a": X, "b": X, "c": Z})
+pw3 = PauliWord({0: Z, "b": Z, "c": Z})
+pw4 = PauliWord({})
+
+ps1 = PauliSentence({pw1: 1.23, pw2: 4j, pw3: -0.5})
+ps2 = PauliSentence({pw1: -1.23, pw2: -4j, pw3: 0.5})
+ps3 = PauliSentence({pw3: -0.5, pw4: 1})
+ps4 = PauliSentence({pw4: 1})
+ps5 = PauliSentence({})
+
+
+class TestPauliWord:
+    def test_identity_removed_on_init(self):
+        """Test that identities are removed on init."""
+        pw = PauliWord({0: I, 1: X, 2: Y})
+        assert 0 not in pw.keys()  # identity ops are removed from pw
+
+    def test_missing(self):
+        """Test the result when a missing key is indexed"""
+        pw = PauliWord({0: I, 1: X, 2: Y})
+        assert 3 not in pw.keys()
+        assert pw[3] == I
+
+    def test_set_items(self):
+        """Test that setting items raises an error"""
+        pw = PauliWord({0: I, 1: X, 2: Y})
+        with pytest.raises(NotImplementedError):
+            pw[3] = Z  # trying to add to a pw after instantiation is prohibited
+
+    def test_update_items(self):
+        """Test that updating items raises an error"""
+        pw = PauliWord({0: I, 1: X, 2: Y})
+        with pytest.raises(NotImplementedError):
+            pw.update({3: Z})  # trying to add to a pw after instantiation is prohibited
+
+    def test_hash(self):
+        """Test that a unique hash exists for different PauliWords."""
+        pw_1 = PauliWord({0: I, 1: X, 2: Y})
+        pw_2 = PauliWord({0: I, 1: X, 2: Y})  # same as 1
+        pw_3 = PauliWord({1: X, 2: Y, 0: I})  # same as 1 but reordered
+        pw_4 = PauliWord({1: Z, 2: Z})  # distinct from above
+
+        assert pw_1.__hash__() == pw_2.__hash__()
+        assert pw_1.__hash__() == pw_3.__hash__()
+        assert pw_1.__hash__() != pw_4.__hash__()
+
+    @pytest.mark.parametrize("pw", (pw1, pw2, pw3, pw4))
+    def test_copy(self, pw):
+        """Test that the copy is identical to the original."""
+        copy_pw = copy(pw)
+        assert copy_pw == pw
```

Comment:
Check that is not the same instance
```suggestion
        assert copy_pw == pw
        assert copy_pw is not pw
```